pypesto/visualize/model_fit.py

```diff
@@ -174,24 +176,72 @@ def time_trajectory_model(
     if timepoints is None:
         end_time = max(problem.objective.edatas[0].getTimepoints())
         timepoints = np.linspace(start=0, stop=end_time, num=n_timepoints)
-    obj = problem.objective.set_custom_timepoints(timepoints_global=timepoints)
 
-    # evaluate objective with return dic = True to get data
+    # get optimization result
     parameters = result.optimize_result.list[start_index]['x']
+
     # reduce vector to only include free indices. Needed downstream.
     parameters = problem.get_reduced_vector(parameters)
-    ret = obj(parameters, mode='mode_fun', sensi_orders=(0,), return_dict=True)
+
+    # simulate with custom timepoints for hierarchical model
+    if isinstance(problem.objective.calculator, HierarchicalAmiciCalculator):
+        # get parameter dictionary
+        x_dct = dict(
+            zip(problem.x_names, result.optimize_result.list[start_index].x)
+        )
+
+        # evaluate objective with return dict = True to get inner parameters
+        ret = problem.objective(
+            parameters, mode='mode_fun', sensi_orders=(0,), return_dict=True
+        )
+
+        # update parameter dictionary with inner parameters
+        inner_parameters = ret['inner_parameters']
+        x_dct.update(inner_parameters)
+
+        parameter_mapping = problem.objective.parameter_mapping
+        edatas = copy.deepcopy(problem.objective.edatas)
+        amici_model = problem.objective.amici_model
+        amici_solver = problem.objective.amici_solver
+
+        for j in range(len(edatas)):
+            edatas[j].setTimepoints(timepoints)
+
+        amici.parameter_mapping.fill_in_parameters(
+            edatas=edatas,
+            problem_parameters=x_dct,
+            scaled_parameters=True,
+            parameter_mapping=parameter_mapping,
+            amici_model=amici_model,
+        )
+
+        rdatas = amici.runAmiciSimulations(
+            amici_model,
+            amici_solver,
```

Comment:
Depending on the settings on amici_solver, this might compute sensitivities. Create copy of amici_solver and disable computing sensitivities?

Reply:
Does the computation of sensitivities create any harm? Or why would it be desirable not to compute them?

Reply:
For non-small problems, the extra computation time is potentially quite annoying.